lib/mayaHydra/hydraExtensions/CMakeLists.txt

```
@@ -30,31 +30,54 @@ set(HEADERS
 # -----------------------------------------------------------------------------
 # compiler configuration
 # -----------------------------------------------------------------------------
-target_compile_definitions(${TARGET_NAME} 
-    PRIVATE 
-        MAYAHYDRALIB_EXPORT
-        $<$<BOOL:${IS_MACOSX}>:OSMac_>
-        # Copy-pasted from lib/mayaUsd/CMakeLists.txt
-)
-
+if(DEFINED MAYAUSD_LOCATION)
+    target_compile_definitions(${TARGET_NAME} 
+        PRIVATE 
+            MAYAHYDRALIB_EXPORT
+            MAYAHYDRALIB_MAYAUSD_IS_USED_TO_BUILD
+            $<$<BOOL:${IS_MACOSX}>:OSMac_>
+            # Copy-pasted from lib/mayaUsd/CMakeLists.txt
+    )
+else()
+    target_compile_definitions(${TARGET_NAME} 
+        PRIVATE 
+            MAYAHYDRALIB_EXPORT
+            $<$<BOOL:${IS_MACOSX}>:OSMac_>
+            # Copy-pasted from lib/mayaUsd/CMakeLists.txt
+    )
+endif()
+        
 mayaHydra_compile_config(${TARGET_NAME})
 
-if(NOT BUILD_MAYAUSD_LIBRARY)
-    # -----------------------------------------------------------------------------
-    # include directories
-    # -----------------------------------------------------------------------------
-    target_include_directories(${TARGET_NAME} 
-        PUBLIC
-            ${MAYA_INCLUDE_DIRS}
-            ${PYTHON_INCLUDE_DIR}
-            ${PXR_INCLUDE_DIRS}
-            ${BOOST_INCLUDE_DIR}
-            ${CMAKE_BINARY_DIR}/include
-        PRIVATE
-            $<$<BOOL:${UFE_FOUND}>:${UFE_INCLUDE_DIR}>
-    )
+# -----------------------------------------------------------------------------
+# include directories
+# -----------------------------------------------------------------------------
+if(DEFINED MAYAUSD_LOCATION)
+target_include_directories(${TARGET_NAME} 
+            PUBLIC
+                ${MAYA_INCLUDE_DIRS}
+                ${PYTHON_INCLUDE_DIR}
+                ${PXR_INCLUDE_DIRS}
+                ${BOOST_INCLUDE_DIR}
+                ${CMAKE_BINARY_DIR}/include
+        ${MAYAUSD_LOCATION}/include
+            PRIVATE
+                $<$<BOOL:${UFE_FOUND}>:${UFE_INCLUDE_DIR}>
+)
+     else()
+        target_include_directories(${TARGET_NAME} 
+            PUBLIC
+                ${MAYA_INCLUDE_DIRS}
+                ${PYTHON_INCLUDE_DIR}
+                ${PXR_INCLUDE_DIRS}
+                ${BOOST_INCLUDE_DIR}
+                ${CMAKE_BINARY_DIR}/include
+            PRIVATE
+                $<$<BOOL:${UFE_FOUND}>:${UFE_INCLUDE_DIR}>
+        )
+     endif()
```

Can simplify with generator expression.
```suggestion
    target_include_directories(${TARGET_NAME} 
        PUBLIC
            ${MAYA_INCLUDE_DIRS}
            ${PYTHON_INCLUDE_DIR}
            ${PXR_INCLUDE_DIRS}
            ${BOOST_INCLUDE_DIR}
            ${CMAKE_BINARY_DIR}/include
        PRIVATE
            $<$<BOOL:${UFE_FOUND}>:${UFE_INCLUDE_DIR}>
            $<$<BOOL:${MAYAUSD_FOUND}>:${MAYAUSD_INCLUDE_DIR}>
    )
```